packages/material-ui/src/RadioGroup/RadioGroup.test.js

```diff
@@ -274,16 +274,31 @@ describe('<RadioGroup />', () => {
   });
 
   describe('useRadioGroup', () => {
-    const RadioGroupController = React.forwardRef((_, ref) => {
+    const RadioGroupItem = React.forwardRef(function RadioGroupItem(_, ref) {
       const radioGroup = useRadioGroup();
-      React.useImperativeHandle(ref, () => radioGroup, [radioGroup]);
-      return null;
+
+      const hostRef = React.useRef(null);
+      React.useImperativeHandle(
+        ref,
+        () => {
+          return {
+            ...radioGroup,
+            // `name` might be an opaque identifier so we want the "materialized" value
+            get name() {
+              return hostRef.current.getAttribute('data-name');
```

Comment:
Not concurrent safe, need to rethink. Probably better to return the opaque identifier.